### ISSUE TITLE
Option to implicitly end ConversationHandler

### ISSUE THREAD

**Issue by loozhengyuan** [python-telegram-bot/python-telegram-bot#1365]
Perhaps I should start off by sharing the problem I am facing. As of now, in any conversation, I am able to call other slash commands (depending on whether the ConversationHandler comes before or after that other command). With this, I was able to create sort of like a "nested conversations" scenario. 

To circumvent the above phenomenon, there are currently fallbacks which I can use, but requires the user to **explicitly** trigger (e.g. /done) to return ConversationHandler.END. However, just like in natural conversation, I envision users to be able to trigger new conversations automatically without having to explicitly end the current. 

Possible ways this can be done:
1. Option in all handlers (or maybe just ConversationHandler) to check if a conversation already exists, if so then return ConversationHandler.END before proceeding.
2. Option in ConversationHandler to end conversation if other commands are called. (Not sure how this is going to work in terms of the logic)

That said, I am not aware if there's anyone that designs their bot in such a way that nests conversations, so i guess have an option flag may be good?

Any comments/feedback are welcome. Or if there's a way I can circumvent my problem that I overlooked, please do let me know too. Thank you!

**Comment by jsmnbom** [comment]
A bit of a hacky workaround would be something like

```python
def fallback(update, context):
  context.update_queue.append(update)
  return ConversationHandler.END

# in your conversationhandler
fallbacks = [MessageHandler(Filters.command, fallback)]

```
I might have missed something, but that sorta sounds what you're after - note that it might have a slight chance of a race condition, but there are probably ways of getting around that too.

**Comment by Poolitzer** [comment]
The workaround seems to be a good solution, nested Conversations might be implemented via #1153.

Another solution could be adding conversationhandlers in different groups, and giving the entries the same commands as the fallbacks (with conversation_handler ending). This could work as well. I will close this issue for now :)
